### DISCUSSION TITLE
coexistence of python and bash: strange beahviour

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#3700] [Q&A] [answered]
Hi,
using this code

~~~~
---
title: "bash"
format: html
---

# hello

```{bash}
curl --version | head -n 1 | awk '{ print $2 }'
```
~~~~

I have 

![image](https://user-images.githubusercontent.com/30607/208264706-af89c332-1f30-44dc-8688-b70f8ba72783.png)

If I add some python code

~~~~
---
title: "bash"
format: html
---

# hello

```{bash}
curl --version | head -n 1 | awk '{ print $2 }'
```

```{python}
print("hello")
```
~~~~

the python code is not executed but simply printed

![image](https://user-images.githubusercontent.com/30607/208264806-e90183b6-1de7-4038-8fe3-6a5b19d484af.png)

If I change the order of code blocks, the python code works, and the bash code is simply printed.

If I force the engine in this way - `engine: knitr` - both blocks work properly.

![image](https://user-images.githubusercontent.com/30607/208264908-dedaef2f-f08b-4e72-ac24-f06ac37cbe8b.png)

Is it normal? In the [engine binding page](https://quarto.org/docs/computations/execution-options.html#engine-binding) I read that in some way the jupyter engine is the one to make python and bash coexist, but if I use  jupyter engine, I can't get them to coexist.

Do I must use knitr to do it? cc @jjallaire 

Thank you

**Comment by jjallaire** [comment]
Bash and Python can coexist as you have them here only in Knitr.

If on the other hand you use Jupyter shell magics (https://jakevdp.github.io/PythonDataScienceHandbook/01.05-ipython-and-shell-commands.html) you can do something similar in Jupyter (but not if this is quite as general).

**Reply by aborruso** [reply]
Thank you, I have created a related PR https://github.com/quarto-dev/quarto-web/compare/main...aborruso:quarto-web:patch-4

**Comment by jjallaire** [comment, marked as answer]
Here is another refinement of the docs: https://quarto.org/docs/computations/execution-options.html#shell-commands

**Reply by aborruso** [reply]
Thank you very much
